packages/swing-store/src/snapStore.js

```diff
@@ -161,7 +164,9 @@ export function makeSnapStore(
     for (const hash of toDelete) {
       const fullPath = hashPath(hash);
       try {
-        unlinkSync(fullPath);
+        if (keepSnapshots !== true) {
```

Comment:
that seems reasonable, it's not something that could be turned on and off over time (we don't persist the list of deletable files, so once `commitDeletes` is called, we forget entirely about the deletability of each file), but I don't think that's a problem for this use case

Reply:
Yeah I double checked what happens if the snap-store tries to write a file which already exists.

There is first a `existsSync` check, so if the file exists then, it simply doesn't try to write the file to disk.
https://github.com/Agoric/agoric-sdk/blob/6747ffc6f520990cec223302a65b1551cb44cdc2/packages/swing-store/src/snapStore.js#L127-L129

Then no matter what, `atomicWrite` uses [`fs.rename`](https://nodejs.org/docs/latest-v16.x/api/fs.html#fsrenameoldpath-newpath-callback), which overwrites the destination in case it already existed. So there is no way this could fail.
https://github.com/Agoric/agoric-sdk/blob/6747ffc6f520990cec223302a65b1551cb44cdc2/packages/swing-store/src/snapStore.js#L79